lib/galaxy/jobs/runners/__init__.py

```diff
@@ -450,6 +451,7 @@ def __init__(self, job_wrapper, job_destination):
         self.runner_state_handled = False
         self.job_wrapper = job_wrapper
         self.job_destination = job_destination
+        self.redact_email_in_job_name = self.job_wrapper.app.config.redact_email_in_job_name
```

Comment:
the job_wrapper presence seem to not be guaranteed?
```
galaxy.jobs.runners ERROR 2018-05-11 12:55:41,351 (3) Unhandled exception calling queue_job
Traceback (most recent call last):
  File "/galaxy/lib/galaxy/jobs/runners/__init__.py", line 113, in run_next
    method(arg)
  File "/galaxy/lib/galaxy/jobs/runners/pulsar.py", line 312, in queue_job
    pulsar_job_state = AsynchronousJobState()
  File "/galaxy/lib/galaxy/jobs/runners/__init__.py", line 503, in __init__
    super(AsynchronousJobState, self).__init__(job_wrapper, job_destination)
  File "/galaxy/lib/galaxy/jobs/runners/__init__.py", line 454, in __init__
    self.redact_email_in_job_name = self.job_wrapper.app.config.redact_email_in_job_name
AttributeError: 'NoneType' object has no attribute 'app'
```

source: https://jenkins.galaxyproject.org/job/docker-integration/5130/console